internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,142 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) (catalogdEntitySource, error) {
+	entities := catalogdEntitySource{}
+	bundleMetadatas, packageMetdatas, err := onClusterMetadata(ctx, client)
```

Comment:
> `onCluster` as opposed to what? Why is that meaningful?

As opposed to [static hardcoded entitySource](https://github.com/operator-framework/operator-controller/blob/a252719a953c59063947977f88d8bdbcde96887f/main.go#L94). I'm open to a better name suggestion though, but like I mentioned, this code is definitely subject to change soon

Reply:
If this only has meaning as compared to some hardcoded thing being ripped out right now I'd just say leave the extra adjectives out.

Reply:
Agree with @everettraven. I think the `New` function should return a struct that has a single `client.Client` field. Then we use that client during calls to the entity source functions to actually fetch the metadata on the fly. This theoretically ensures we get up-to-date metadata every time (and yes, I realize our existing CRD approach means there are probably some holes in that theory, but I'm not terribly concerned since we know we're going back to that API in a future milestone anyway).